pkg/operator/operator.go

```diff
@@ -401,6 +412,39 @@ func New(
 	)
 	o.informerFactories = append(o.informerFactories, kubeInformersOperatorNS)
 
+	configInformers := configv1informers.NewSharedInformerFactory(configClient, 10*time.Minute)
+	missingVersion := "0.0.1-snapshot"
+	desiredVersion := missingVersion
+	if envVersion, exists := os.LookupEnv("RELEASE_VERSION"); exists {
+		desiredVersion = envVersion
+	}
+
+	// By default, this will exit(0) if the featuregates change
+	featureGateAccessor := featuregates.NewFeatureGateAccess(
+		desiredVersion, missingVersion,
+		configInformers.Config().V1().ClusterVersions(),
+		configInformers.Config().V1().FeatureGates(),
+		eventRecorder,
+	)
+	go featureGateAccessor.Run(ctx)
+	go configInformers.Start(ctx.Done())
+
+	select {
+	case <-featureGateAccessor.InitialFeatureGatesObserved():
+		featureGates, _ := featureGateAccessor.CurrentFeatureGates()
+		klog.Infof("FeatureGates initialized: %v", featureGates.KnownFeatures())
+	case <-time.After(1 * time.Minute):
+		klog.Errorf("timed out waiting for FeatureGate detection")
+		return nil, fmt.Errorf("timed out waiting for FeatureGate detection")
+	}
+
+	featureGates, err := featureGateAccessor.CurrentFeatureGates()
+	if err != nil {
+		return nil, err
+	}
+	// read featuregate read and usage to set a variable to pass to a controller
+	o.metricsServerEnabled = featureGates.Enabled(configv1.FeatureGateMetricsServer)
```

Comment:
It might be good to have an info log here when the metrics server gate is enabled:
```suggestion
	o.metricsServerEnabled = featureGates.Enabled(configv1.FeatureGateMetricsServer)
	if o.metricsServerEnabled {
		klog.Info("Metrics server enabled")
	}
```